src/channel.cpp

```diff
@@ -511,6 +513,24 @@ void CChannel::Disconnect()
         // (assuming that no audio packet is received in the meantime)
         iConTimeOut = 1; // a small number > 0
     }
+    
+    if ( !bIsServer )
+    {
+        if ( IsConnected() )
+        {
+            // for a Client, block further audio data and disable the channel as soon as Disconnect() is called
+            // TODO: Add reasoning from #2550
+            bDisconnectAndDisable = true;
+        }
+        else
+        {
+            // For disconnected clients set defaults
```

Comment:
Why isn't this done when the disconnect _actually_ happens, rather than here?  Like in line 645?  Is this just duplicated?  Or is there a reason for it?

Reply:
I think there's a reason for it. 
It's probably trying to ensure that bDisconnectAndDisable is representing the client being in process of disconnecting?

Reply:
I don't see what job it's doing that setting `iConTimeOut` to 1 doesn't already do.